### PR TITLE
[tNJiNQW5] Converted to Swift 4

### PR DESCRIPTION
**Pull request by Kirillzzy**
**Тема ревью**
Конвертация под Swift 4

**Описание ревью**
Конвертировал приложение под Swift 4.
Исправил ворнинги Swift'a и Swiftlint'a.

**На что обратить внимание и как тестировать**
Запустить приложение и убедиться, что ничего не сломалось

**Связанные таски**
https://trello.com/c/tNJiNQW5/392-convert-to-swift-4

### CODE REVIEW

File: CHMeetupApp/Sources/ViewControllers/Events/Registration/Preview/DisplayModel/RegistrationFormDataCollection.swift [hunk 1]
```diff
@@ -70,12 +70,13 @@ final class FormDisplayCollection: NSObject, DisplayCollection, DisplayCollectio
 
   func headerTitle(for section: Int) -> NSAttributedString {
     let cell = formData.sections[section]
```
Comment (anatoliykant) [inline CHMeetupApp/Sources/ViewControllers/Events/Registration/Preview/DisplayModel/RegistrationFormDataCollection.swift:72]: let attribute = [NSAttributedStringKey.font: DefaultTableHeaderView.font]

File: CHMeetupApp/Sources/ViewControllers/Events/Registration/Preview/DisplayModel/RegistrationFormDataCollection.swift [hunk 2]
```diff
@@ -70,12 +70,13 @@ final class FormDisplayCollection: NSObject, DisplayCollection, DisplayCollectio
...
-    let attributtedString = NSMutableAttributedString(string: cell.name,
-                                                      attributes: [NSFontAttributeName: DefaultTableHeaderView.font])
+    let attributtedString
+      = NSMutableAttributedString(string: cell.name,
+                                  attributes: [NSAttributedStringKey.font: DefaultTableHeaderView.font])
```
Comment (anatoliykant) [inline CHMeetupApp/Sources/ViewControllers/Events/Registration/Preview/DisplayModel/RegistrationFormDataCollection.swift]: attributes: attribute)

Comment (anatoliykant) [reply CHMeetupApp/Sources/ViewControllers/Events/Registration/Preview/DisplayModel/RegistrationFormDataCollection.swift]: SwiftLint warning: Line should be 120 characters or less: currently 124 characters

File: CHMeetupApp/Sources/Model/Entities/EventEntity.swift [hunk 3]
```diff
@@ -49,6 +49,7 @@ final class EventEntity: TemplatableObject, TemplateEntity {
     }
   }
 
+  @objc dynamic var id: Int = 0
```
Comment (anatoliykant) [inline CHMeetupApp/Sources/Model/Entities/EventEntity.swift]: move to 61 lines

File: CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift [hunk 4]
```diff
@@ -15,16 +15,16 @@ struct MethodSwizzler {
 
     let methodAdded = class_addMethod(objectClass,
                                       originalSelector,
```
Comment (anatoliykant) [inline CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift]: Line 13:
**guard** let originalMethod = class_getInstanceMethod(objectClass, originalSelector) **else { return }**
Line 14:
**guard** let swizzledMethod = class_getInstanceMethod(objectClass, swizzledSelector) **else { return }**

File: CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift [hunk 5]
```diff
@@ -15,16 +15,16 @@ struct MethodSwizzler {
...
-                                      method_getImplementation(swizzledMethod),
-                                      method_getTypeEncoding(swizzledMethod))
+                                      method_getImplementation(swizzledMethod!),
```
Comment (anatoliykant) [inline CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift]: delete ! - force unwrap

File: CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift [hunk 6]
```diff
@@ -15,16 +15,16 @@ struct MethodSwizzler {
...
+                                      method_getTypeEncoding(swizzledMethod!))
```
Comment (anatoliykant) [inline CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift]: delete ! - force unwrap

File: CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift [hunk 7]
```diff
@@ -15,16 +15,16 @@ struct MethodSwizzler {
...
 
     if methodAdded {
       class_replaceMethod(objectClass,
                           swizzledSelector,
-                          method_getImplementation(originalMethod),
-                          method_getTypeEncoding(originalMethod))
+                          method_getImplementation(originalMethod!),
```
Comment (anatoliykant) [inline CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift]: delete ! - force unwrap

File: CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift [hunk 8]
```diff
@@ -15,16 +15,16 @@ struct MethodSwizzler {
...
+                          method_getTypeEncoding(originalMethod!))
```
Comment (anatoliykant) [inline CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift]: delete ! - force unwrap

File: CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift [hunk 9]
```diff
@@ -15,16 +15,16 @@ struct MethodSwizzler {
...
     } else {
-      method_exchangeImplementations(originalMethod, swizzledMethod)
+      method_exchangeImplementations(originalMethod!, swizzledMethod!)
```
Comment (anatoliykant) [inline CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift]: delete two ! - force unwrap

File: CHMeetupApp/Sources/Model/Entities/UserEntity.swift [hunk 10]
```diff
@@ -10,31 +10,30 @@ import Foundation
 import RealmSwift
 
 final class UserEntity: TemplatableObject, TemplateEntity {
-
+  @objc dynamic var id: Int = 0
```
Comment (anatoliykant) [inline CHMeetupApp/Sources/Model/Entities/UserEntity.swift]: move to line 20

File: CHMeetupApp/Sources/Model/Entities/UserEntity.swift [hunk 11]
```diff
@@ -10,31 +10,30 @@ import Foundation
...
+  @objc dynamic var remoteId: Int = 0
```
Comment (anatoliykant) [inline CHMeetupApp/Sources/Model/Entities/UserEntity.swift]: move to line 21

File: CHMeetupApp/Sources/Common/Extensions/Foundation/Dictionary/Dictionary+HTTPBody.swift [hunk 12]
```diff
@@ -13,8 +13,8 @@ extension Dictionary where Key: ExpressibleByStringLiteral, Value: ExpressibleBy
   var httpQuery: Data {
     var httpQuery: [String] = []
     self.forEach { key, value in
-      let keyString = String("\(key)")?.urlEncoding ?? ""
-      let valueString = String("\(value)")?.urlEncoding ?? ""
+      let keyString = "\(key)".urlEncoding
```
Comment (anatoliykant) [inline CHMeetupApp/Sources/Common/Extensions/Foundation/Dictionary/Dictionary+HTTPBody.swift:15]: Is it possible to nil?

File: CHMeetupApp/Sources/Common/Extensions/Foundation/Dictionary/Dictionary+HTTPBody.swift [hunk 13]
```diff
@@ -13,8 +13,8 @@ extension Dictionary where Key: ExpressibleByStringLiteral, Value: ExpressibleBy
...
+      let valueString = "\(value)".urlEncoding
```
Comment (anatoliykant) [inline CHMeetupApp/Sources/Common/Extensions/Foundation/Dictionary/Dictionary+HTTPBody.swift:16]: Is it possible to nil?

File: CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift [hunk 14]
```diff
@@ -10,12 +10,12 @@ import Foundation
 
 extension String {
   func substring(from index: Int) -> String {
-    return substring(from: self.index(startIndex, offsetBy: index))
+    return String(self[self.index(startIndex, offsetBy: index)...])
```
Comment (anatoliykant) [inline CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift:13]: let ind = self.index(startIndex, offsetBy: index)
return String(self[ind...])

easier to read 👆

File: CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift [hunk 15]
```diff
@@ -10,12 +10,12 @@ import Foundation
...
   }
 
   func substring(with range: Range<Int>) -> String {
-    let startIndex = self.index(self.startIndex, offsetBy: range.lowerBound)
-    let endIndex = self.index(self.startIndex, offsetBy: range.upperBound)
-    return substring(with: startIndex ..< endIndex)
+    let startIndex = index(self.startIndex, offsetBy: range.lowerBound)
+    let endIndex = index(self.startIndex, offsetBy: range.upperBound)
+    return String(self[startIndex..<endIndex])
```
Comment (anatoliykant) [inline CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift:19]: return String(self[startInd..<endInd])

File: CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift [hunk 16]
```diff
@@ -10,12 +10,12 @@ import Foundation
...
-    return substring(from: self.index(startIndex, offsetBy: index))
+    return String(self[self.index(startIndex, offsetBy: index)...])
   }
 
   func substring(with range: Range<Int>) -> String {
-    let startIndex = self.index(self.startIndex, offsetBy: range.lowerBound)
-    let endIndex = self.index(self.startIndex, offsetBy: range.upperBound)
-    return substring(with: startIndex ..< endIndex)
+    let startIndex = index(self.startIndex, offsetBy: range.lowerBound)
+    let endIndex = index(self.startIndex, offsetBy: range.upperBound)
```
Comment (anatoliykant) [inline CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift:18]: let endInd = index(startIndex, offsetBy: range.upperBound)

File: CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift [hunk 17]
```diff
@@ -10,12 +10,12 @@ import Foundation
...
   func substring(from index: Int) -> String {
-    return substring(from: self.index(startIndex, offsetBy: index))
+    return String(self[self.index(startIndex, offsetBy: index)...])
   }
 
   func substring(with range: Range<Int>) -> String {
-    let startIndex = self.index(self.startIndex, offsetBy: range.lowerBound)
-    let endIndex = self.index(self.startIndex, offsetBy: range.upperBound)
-    return substring(with: startIndex ..< endIndex)
+    let startIndex = index(self.startIndex, offsetBy: range.lowerBound)
```
Comment (anatoliykant) [inline CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift:17]: let startInd = index(startIndex, offsetBy: range.lowerBound)

**Review by anatoliykant — APPROVED** [review]
NotificationHelper.swift
line 25 - SwiftLint warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)

**Comment by anatoliykant** [comment]
I suggest moving the [Run Script Swiftlint](https://www.dropbox.com/s/uo5p8z2cfq05ygh/Swiftlint%20script.png?dl=0) right after the Run Script Natalie and above the Compile sources.
To check the style of the code was before the direct compilation of the project.

**Comment by smejl** [comment]
@Kirillzzy, when you can fix comments?

**Comment by Kirillzzy** [comment]
@smejl, today I guess. Thanks for remind)

File: CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift (hunk 15 above)
Comment (Kirillzzy) [reply CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift:19]: You know, we don't shorten vars and other names.

File: CHMeetupApp/Sources/Common/Extensions/Foundation/Dictionary/Dictionary+HTTPBody.swift (hunk 13 above)
Comment (Kirillzzy) [reply CHMeetupApp/Sources/Common/Extensions/Foundation/Dictionary/Dictionary+HTTPBody.swift:16]: No, it isn't an Optional type.

File: CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift (hunk 14 above)
Comment (smejl) [reply CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift:13]:
```
return String(self[index(startIndex, offsetBy: index)...])
```

File: CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift (hunk 17 above)
Comment (smejl) [reply CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift:17]: Remove self, please.

File: CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift (hunk 16 above)
Comment (smejl) [reply CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift:18]: Here too.

File: CHMeetupApp/Sources/Common/Helpers/DisplayCollectionDelegate.swift [hunk 18]
```diff
@@ -25,7 +25,7 @@ extension UIViewController: DisplayCollectionDelegate {
     navigationController?.pushViewController(viewController, animated: true)
   }
 
-  func updateUI() {
+  @objc func updateUI() {
```
Comment (smejl) [inline CHMeetupApp/Sources/Common/Helpers/DisplayCollectionDelegate.swift:28]:
```
@objc 
func updateUI() {
```

File: CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift [hunk 19]
```diff
@@ -10,14 +10,12 @@ import Foundation
 
 struct MethodSwizzler {
   static func swizzleMethods(objectClass: AnyClass, originalSelector: Selector, swizzledSelector: Selector) {
-    let originalMethod = class_getInstanceMethod(objectClass, originalSelector)
-    let swizzledMethod = class_getInstanceMethod(objectClass, swizzledSelector)
-
+    guard let originalMethod = class_getInstanceMethod(objectClass, originalSelector),
+      let swizzledMethod = class_getInstanceMethod(objectClass, swizzledSelector) else { return }
```
Comment (smejl) [inline CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift:14]: Write else block in new line
```
 else { 
    return 
}
```

File: CHMeetupApp/Sources/Common/Views/PlateTableViewCell.swift [hunk 20]
```diff
@@ -35,7 +35,7 @@ class PlateTableViewCell: UITableViewCell {
   }
 
   // For UIAppearance proxy
-  dynamic var plateAppearance: PlateTableViewCellAppearance? {
+  @objc dynamic var plateAppearance: PlateTableViewCellAppearance? {
```
Comment (smejl) [inline CHMeetupApp/Sources/Common/Views/PlateTableViewCell.swift]: Add line break.

File: CHMeetupApp/Sources/ViewControllers/Base/AZTabBarController.swift [hunk 21]
```diff
@@ -161,7 +161,7 @@ public class AZTabBar: UITabBar {
 }
 
 extension UIViewController {
-  open func az_tabBarItemContentView() -> AZTabBarItemView {
+  @objc open func az_tabBarItemContentView() -> AZTabBarItemView {
```
Comment (smejl) [inline CHMeetupApp/Sources/ViewControllers/Base/AZTabBarController.swift:164]: Line break.

File: CHMeetupApp/Sources/ViewControllers/Events/EventPreview/EventPreviewViewController.swift [hunk 22]
```diff
@@ -82,14 +82,14 @@ class EventPreviewViewController: UIViewController {
     }
   }
 
-  func acceptAction() {
+  @objc func acceptAction() {
```
Comment (smejl) [inline CHMeetupApp/Sources/ViewControllers/Events/EventPreview/EventPreviewViewController.swift:85]: Use line break for function, when you set attributes, for example, @objc or @discardableresult

**Comment by smejl** [comment]
@Kirillzzy, thanks. Fix please minor comments and we can merge your branch.

File: CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift (hunk 14 above)
Comment (Kirillzzy) [reply CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift:13]: No)) index here is local, but I need self.

File: CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift (hunk 16 above)
Comment (Kirillzzy) [reply CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift:18]: Noo, the answer above

File: CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift (hunk 17 above)
Comment (Kirillzzy) [reply CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift:17]: Noo)

File: CHMeetupApp/Sources/Common/Helpers/DisplayCollectionDelegate.swift (hunk 18 above)
Comment (Kirillzzy) [reply CHMeetupApp/Sources/Common/Helpers/DisplayCollectionDelegate.swift:28]: ??

File: CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift (hunk 19 above)
Comment (Kirillzzy) [reply CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift:14]: It doesn't matter)

File: CHMeetupApp/Sources/ViewControllers/Events/EventPreview/EventPreviewViewController.swift (hunk 22 above)
Comment (Kirillzzy) [reply CHMeetupApp/Sources/ViewControllers/Events/EventPreview/EventPreviewViewController.swift:85]: maan)

File: CHMeetupApp/Sources/Common/Helpers/DisplayCollectionDelegate.swift (hunk 18 above)
Comment (smejl) [reply CHMeetupApp/Sources/Common/Helpers/DisplayCollectionDelegate.swift:28]: Line break. Commented below.

File: CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift (hunk 19 above)
Comment (smejl) [reply CHMeetupApp/Sources/Common/Helpers/MethodSwizzler.swift:14]: It does matter for code style

File: CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift (hunk 14 above)
Comment (smejl) [reply CHMeetupApp/Sources/Common/Extensions/Foundation/String/String+Range.swift:13]: You are right. Sorry.

**Review by smejl — APPROVED** [review]

**Pull request merged**